lib/array_equals.rb

```diff
@@ -1,5 +1,26 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  i = 0
+
+  if (array1 == nil && array2 == nil) || (array1 == [] && array2 == [])
```

Comment:
Good boundary checking!

Super-tiny nit: it's _somewhat_ explicit though - could we do something more concise, like:
```
 if (array1 == array2)
   return true
 end
```

(Aside: some people think explicit code is easier to read, while
others prefer more concise code - so this comment is really a 
matter of personal preference. :smile: )